object-shorthand rule and arrow functions

I'm trying out the new [object-shorthand rule](http://eslint.org/docs/rules/object-shorthand) and have run into a snag in my codebase. We have a number of places where we are using arrow functions as object properties, such as when they are passed in an options object to a function.

For example:

```js
myFunction({
  onComplete: () => this._handleComplete(),
});
```

I would like to be able to turn on this rule but allow arrow functions. I didn't find anything in the documentation for the rule that would enable this.

What do you think?

Sure!

With the following file:

```js
function myFunction(options) {
  if (options.onComplete) {
    options.onComplete();
  }
}

myFunction({
  onComplete: () => this._handleComplete(),
});
```

And the following config:

```yaml
object-shorthand:
  - 2
  - methods
```

ESLint will report the following:

```
myfile.js
  8:2  error  Expected method shorthand

✖ 1 problem (1 error, 0 warnings)
```

--- a/docs/rules/object-shorthand.md
+++ b/docs/rules/object-shorthand.md
@@ -60,6 +60,15 @@ var foo = {
 };
 ```
 
+This rule does not flag arrow functions inside of object literals.
+The following will *not* warn:
+
+```js
+var foo = {
+    x: (y) => y
+};
+```
+
 ### Options
 
 The rule takes an option which specifies when it should be applied. It can be set to

--- a/lib/rules/object-shorthand.js
+++ b/lib/rules/object-shorthand.js
@@ -44,11 +44,7 @@ module.exports = function(context) {
                 return;
             }
 
-            if (node.value.type === "ArrowFunctionExpression" && APPLY_TO_METHODS) {
-
-                // {x: ()=>{}} should be written as {x() {}}
-                context.report(node, "Expected method shorthand.");
-            } else if (node.value.type === "FunctionExpression" && APPLY_TO_METHODS) {
+            if (node.value.type === "FunctionExpression" && APPLY_TO_METHODS) {
 
                 // {x: function(){}} should be written as {x() {}}
                 context.report(node, "Expected method shorthand.");

--- a/tests/lib/rules/object-shorthand.js
+++ b/tests/lib/rules/object-shorthand.js
@@ -53,6 +53,12 @@ eslintTester.addRuleTest("lib/rules/object-shorthand", {
         { code: "doSomething({x: y, y() {}})", ecmaFeatures: features},
         { code: "doSomething({y() {}, z: a})", ecmaFeatures: features},
 
+        // arrows functions are still alright
+        { code: "var x = {y: (x)=>x}", ecmaFeatures: features },
+        { code: "doSomething({y: (x)=>x})", ecmaFeatures: features },
+        { code: "var x = {y: (x)=>x, y: a}", ecmaFeatures: features },
+        { code: "doSomething({x, y: (x)=>x})", ecmaFeatures: features },
+
         // options
         { code: "var x = {y() {}}", ecmaFeatures: features, args: [2, "methods"] },
         { code: "var x = {x, y() {}, a:b}", ecmaFeatures: features, args: [2, "methods"] },
@@ -68,7 +74,6 @@ eslintTester.addRuleTest("lib/rules/object-shorthand", {
         { code: "var x = {y: y, x: x}", ecmaFeatures: features, errors: [{ message: "Expected property shorthand.", type: "Property" }, { message: "Expected property shorthand.", type: "Property" }] },
         { code: "var x = {y: z, x: x, a: b}", ecmaFeatures: features, errors: [{ message: "Expected property shorthand.", type: "Property" }] },
         { code: "var x = {y: function() {}}", ecmaFeatures: features, errors: [{ message: "Expected method shorthand.", type: "Property" }] },
-        { code: "var x = {y: (x)=>x}", ecmaFeatures: features, errors: [{ message: "Expected method shorthand.", type: "Property" }] },
         { code: "var x = {y: function*() {}}", ecmaFeatures: features, errors: [{ message: "Expected method shorthand.", type: "Property" }] },
         { code: "var x = {x: y, y: z, a: a}", ecmaFeatures: features, errors: [{ message: "Expected property shorthand.", type: "Property" }] },
         { code: "var x = {x: y, y: z, a: function(){}, b() {}}", ecmaFeatures: features, errors: [{ message: "Expected method shorthand.", type: "Property" }] },
@@ -78,7 +83,6 @@ eslintTester.addRuleTest("lib/rules/object-shorthand", {
         { code: "doSomething({a: 'a', 'x': x})", ecmaFeatures: features, errors: [{ message: "Expected property shorthand.", type: "Property" }] },
         { code: "doSomething({y: function() {}})", ecmaFeatures: features, errors: [{ message: "Expected method shorthand.", type: "Property" }] },
         { code: "doSomething({y: function y() {}})", ecmaFeatures: features, errors: [{ message: "Expected method shorthand.", type: "Property" }] },
-        { code: "doSomething({y: (x)=>x})", ecmaFeatures: features, errors: [{ message: "Expected method shorthand.", type: "Property" }] },
 
         // options
         { code: "var x = {y: function() {}}", ecmaFeatures: features, errors: [{ message: "Expected method shorthand.", type: "Property" }], args: [2, "methods"] },

